### DISCUSSION TITLE
mTLS btween nginx ingress controller and the backend webservers

### DISCUSSION THREAD

**Discussion by mr-elamin** [cilium/cilium#36210] [Q&A]
Hi all,
today I tried to implement the mTLS in the namespace of my ingress nginx controller and the backend namespace where my webservers are living.
I applied the following CNP
![image](https://github.com/user-attachments/assets/2c6a700d-192f-4ac8-9f80-c1199bf7fb4b)
after this, the controller lost its communication with the API server.
I changed the policy to this:
![image](https://github.com/user-attachments/assets/0585a00a-a170-45df-9853-df77be95ae9c)
but still the controller was unable to communicate with the API server,

### E1127 16:13:41.462238       7 leaderelection.go:436] error retrieving resource lock ingress-nginx/ingress-nginx-leader: Get "https://10.96.0.1:443/apis/coordination.k8s.io/v1/namespaces/ingress-nginx/leases/ingress-nginx-leader": http2: client connection lost

any idea?
Thanks

**Comment by wibed** [comment]
i am interested aswell.
how have you created the certificates?

**Comment by mr-elamin** [comment]
According to the Cilium documentation, I don't have to create certificates manually. You can refer to the documentation here:
 [Cilium Mutual Authentication](https://docs.cilium.io/en/latest/network/servicemesh/mutual-authentication/mutual-authentication/).

To enable mutual TLS (mTLS), you just need to enable encryption using IPSec or WireGuard and enable mTLS. This will automatically create one SPIRE server and a per-node SPIRE agent. These components are responsible for creating the certificates and providing SPIFFE identities to the workloads.

The Star Wars tutorial worked successfully, but I encountered issues when trying to secure the connections between the ingress controller and the backend web servers. Things not working correctly.
![image](https://github.com/user-attachments/assets/87bc9516-a017-492d-9b59-2d8f1449eb3c)
